addons/mail/models/mail_template.py

```diff
@@ -89,6 +89,34 @@ def default_get(self, fields):
     can_write = fields.Boolean(compute='_compute_can_write',
                                help='The current user can edit the template.')
 
+    def _geneate_template_for_constrain(self):
+        self.ensure_one()
+        model = self.model_id.model
+        if not model:
+            return
+        record = self.env[model].search([], limit=1)
+        if not record.ids:
+            return
+        mail_template_fields = [
+            'body_html',
+            'subject',
+            'email_cc',
+            'email_from',
+            'email_to',
+            'partner_to'
+        ]
+        return self._generate_template(record.ids, mail_template_fields)
```

Comment:
Ok but what if the template expects some specific values to be passed by context? There are quite a few of these if you grep `ctx` in `mail_template_data.xml`, where we *expect* some value to be passed in because the template is only used in certain ways.

I guess we could make sure all of them have a fallback value but that would break the translations in stable for a small QOL change.

Not a bad idea though, for master